desafio-05/ancogamer/go/func.go

```diff
@@ -0,0 +1,364 @@
+	"sync"
+	"time"
+)
+
+// Area ..
+type Area struct {
+	Codigo string `json:"codigo"`
+	Nome   string `json:"nome"`
+	Qtd    int
+	MediaS float64
+	MaiorS float64
+	MenorS float64
+	//MenorSFuncionarioPointer *Func
+
+	AreaMax string
+	AreaMin string
+}
+
+//Func ..
+type Func struct {
+	ID        int     `json:"id"`
+	Nome      string  `json:"nome"`
+	Sobrenome string  `json:"sobrenome"`
+	Salario   float64 `json:"salario"`
+	Area      string  `json:"area"`
+}
+
+//JSON ..
+type JSON struct {
+	Funcs []Func `json:"funcionarios"`
+	Areas []Area `json:"areas"`
+}
+
+//2
+var (
+	AvgA        string
+	AreasMax    string
+	AreasMin    string
+	c           int
+	LastNameMax string
+)
+
+//SobreS .
+type SobreS struct {
+	LastNameMax  string
+	MaiorS       float64
+	SobreNome    string
+	Funcionarios []*string
+}
+
+//X ..
+func X(bigSalaryByLastName *[]SobreS, dat JSON) {
+	var l int
+	for l = 0; l < len(*bigSalaryByLastName); l++ {
+		if dat.Funcs[c].Sobrenome == (*bigSalaryByLastName)[l].SobreNome {
+			for k := 0; k < len((*bigSalaryByLastName)[l].Funcionarios); k++ {
+				if &dat.Funcs[c].Nome == (*bigSalaryByLastName)[l].Funcionarios[k] {
+					return
+				}
+			}
+			if dat.Funcs[c].Salario > (*bigSalaryByLastName)[l].MaiorS {
+				(*bigSalaryByLastName)[l].MaiorS = dat.Funcs[c].Salario
+				(*bigSalaryByLastName)[l].LastNameMax = ""
+				(*bigSalaryByLastName)[l].LastNameMax = Build((*bigSalaryByLastName)[l].LastNameMax, "\nlast_name_max|", dat.Funcs[c].Sobrenome, "|", dat.Funcs[c].Nome,
+					" ", dat.Funcs[c].Sobrenome, "|",
+					fmt.Sprintf("%.2f", dat.Funcs[c].Salario))
+			} else if dat.Funcs[c].Salario == (*bigSalaryByLastName)[l].MaiorS {
+				(*bigSalaryByLastName)[l].LastNameMax = Build((*bigSalaryByLastName)[l].LastNameMax, "\nlast_name_max|", dat.Funcs[c].Sobrenome, "|", dat.Funcs[c].Nome,
+					" ", dat.Funcs[c].Sobrenome, "|",
+					fmt.Sprintf("%.2f", dat.Funcs[c].Salario))
+			}
+			return
+		}
+	}
+	*bigSalaryByLastName = append(*bigSalaryByLastName, SobreS{
+		LastNameMax: Build("\nlast_name_max|", dat.Funcs[c].Sobrenome, "|", dat.Funcs[c].Nome, " ", dat.Funcs[c].Sobrenome, "|", fmt.Sprintf("%.2f", dat.Funcs[c].Salario)),
+		MaiorS:      dat.Funcs[c].Salario,
+		SobreNome:   dat.Funcs[c].Sobrenome,
+	})
+
+	(*bigSalaryByLastName)[l].Funcionarios = append((*bigSalaryByLastName)[l].Funcionarios, &dat.Funcs[c].Nome)
+
+	return
+
+}
+
+func main() {
+	start := time.Now()
+	/*
+		var optCPUProfile string
+
+		flag.StringVar(&optCPUProfile, "cpuprofile", "", "write cpu profile to file")
+		flag.Parse()
+		if optCPUProfile != "" {
+			println("to aqui")
+			f, err := os.Create(optCPUProfile)
+			if err != nil {
+				log.Fatal(err)
+			}
+			pprof.StartCPUProfile(f)
+			defer pprof.StopCPUProfile()
+		}
+	*/
+
+	ctx, err := ioutil.ReadFile(os.Args[1])
+	if err != nil {
+		log.Fatal(err)
+	}
+	var dat JSON
+	err = json.Unmarshal(ctx, &dat)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	var maiorValor float64 = dat.Funcs[0].Salario
```

Comment:
É uma boa ideia declarar todas essas variávels dentro de um var só. Tipo:
```
var (
  ...
  ...
)
```

Reply:
Voce marcou como resolved mas ainda tem um monte de var horroroso. Vai mudar pra uma declaração só com var (...) ou não? Se não for, explique no comentário porque não.